src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -73,6 +79,11 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
         AspireEventSource.Instance.DcpModelCreationStart();
         try
         {
+            if (!_model.Resources.Any(r => r.Name == DashboardReservedResourceName))
+            {
+                await StartDashboardAsync(cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
This logic should always start the dashboard first. Even if it's a project resource (when debugging locally).

Reply:
The local debug case is addressed in #1775